### PR TITLE
Don't require the application to create rows.

### PR DESCRIPTION
**Pull request by burke**
This gem currently requires us to create a bunch of rows for each shop whose version we're tracking. It seems easier to just create those rows if they're accessed and found to be missing.

Will link to related pull request in shopify shortly.

Review: @hornairs @camilo @jstorimer @boourns

### CODE REVIEW

**Comment by airhorns** [comment]
Is the `ON DUPLICATE KEY` bit to handle the situation where we do the select and nothing comes back, but in the mean time someone else makes the row?

**Comment by boourns** [comment]
Before this change did we have to do a migration to add the rows?  

Or is there somewhere else in the gem where you should remove the creation of all the rows?

**Comment by burke** [comment]
We talked in person, but for the record,

@hornairs: Yes, exactly.

@boourns: Before, the code expected a migration and after_create to populate all the rows before this code was run. Hence, there's nothing to remove.

File: lib/multirow_counter/extension.rb [hunk 1]
```diff
@@ -17,11 +17,19 @@ def multirow_counter(counter_name, options)
...
-        counter_relation.where(:counter_id => randomly_selected_counter_row).limit(1).update_all("value = value+#{Integer(incr)}")
+        val = counter_relation.where(:counter_id => randomly_selected_counter_row).limit(1).update_all("value = value+#{Integer(incr)}")
+
+        if val == 0 # row doesn't exist. We create it instead.
+          counter_relation.connection.execute <<-SQL
+            INSERT INTO #{const.table_name} (#{class_name.foreign_key}, counter_id, value)
+              VALUES (#{id}, #{randomly_selected_counter_row}, #{Integer(incr)})
+              ON DUPLICATE KEY UPDATE value=value+#{Integer(incr)};
+          SQL
+        end
```
Comment (jstorimer) [inline lib/multirow_counter/extension.rb:32]: Nice optimization.

**Comment by jstorimer** [comment]
LGTM

**Pull request merged**